### PR TITLE
feat: revoke a sharing

### PR DESCRIPTION
**Pull request by Ljinod**
This PR is the first step towards a full revokation: the sharing
document is set to revoked and all associated OAuth clients are deleted.

### CODE REVIEW

File: pkg/sharings/sharings.go [hunk 1]
```diff
@@ -597,6 +603,41 @@ func CreateSharing(instance *instance.Instance, sharing *Sharing) error {
 	return couchdb.CreateDoc(instance, sharing)
 }
 
+// RevokeSharing revokes the sharing and deletes all the OAuth client
+// associated with it.
+//
+// Revoking a sharing consists of setting the field `Revoked` to `true`.
+func RevokeSharing(ins *instance.Instance, sharing *Sharing) error {
+	sharing.Revoked = true
```
Comment (nono) [inline pkg/sharings/sharings.go]: You don't persist in db this change?

File: pkg/sharings/sharings.go [hunk 2]
```diff
@@ -597,6 +603,41 @@ func CreateSharing(instance *instance.Instance, sharing *Sharing) error {
...
+	return err
+}
+
+func deleteOAuthClient(ins *instance.Instance, id string) error {
+	client, err := oauth.FindClient(ins, id)
+	if err != nil {
+		log.Errorf("[sharings] Could not find client %s: %v", id, err)
+		return err
+	}
+	crErr := client.Delete(ins)
```
Comment (nono) [inline pkg/sharings/sharings.go]: Why `crErr`? And not just `err = ...`?

File: pkg/sharings/sharings.go (hunk 1 above)
Comment (Ljinod) [inline pkg/sharings/sharings.go]: I should indeed, thanks!

File: pkg/sharings/sharings.go (hunk 2 above)
Comment (Ljinod) [inline pkg/sharings/sharings.go]: Because `client.Delete` returns a `ClientRegistrationError` structure not a simple error

**Pull request merged**